File: examples/objc-demo-app/DemoApp/PlacedOrderEvent.m
#import "PlacedOrderEvent.h"

@implementation PlacedOrderEvent

- (instancetype) initWithCartItems:(NSArray *)cartItems {
    if ((self = [super init])) {
        self._cartItems = cartItems;
    }
    return self;
}

- (NSString *)name {
    return @"placed_order";
}

- (NSDictionary *)parameters {
    NSDictionary *params = [[NSDictionary alloc] init];
    NSNumber *totalPrice = [NSNumber numberWithInteger:0];
    for (int i = 0; i <= self._cartItems.count; i++) {
        CartItem *item = self._cartItems[i];
        [params setValue:item.name forKey: [NSString stringWithFormat: @"item_name_%d", i]];
        [params setValue:item.price forKey: [NSString stringWithFormat: @"item_price_%d", i]];
        [params setValue:item.image forKey: [NSString stringWithFormat: @"item_image_%d", i]];
        totalPrice = [NSNumber numberWithFloat: [item.price floatValue] + [totalPrice floatValue]];
    }
    [params setValue:totalPrice forKey:@"total_price"];
    return params;
}

@end

File: examples/swift-demo-app-with-direct-firebase-dependency/DemoApp/PlacedOrderEvent.swift
import Foundation
import OptimoveSDK
import OptimoveCore

class PlacedOrderEvent: OptimoveEvent {
    
    private let cartItems: [CartItem]
    
    init(_ items: [CartItem]) {
        self.cartItems = items
    }
    
    var name: String {
        return "placed_order"
    }
    
    var parameters: [String : Any] {
        var params: [String: Any] = [:]
        var totalPrice = 0.0
        for i in 0..<self.cartItems.count {
            let item = self.cartItems[i]
            params["item_name_\(i)"] = item.name
            params["item_price_\(i)"] = item.price
            params["item_image_\(i)"] = item.image
            totalPrice += item.price
        }
        params["total_price"] = totalPrice
        return params
    }
}
